Customer API Update V2

emmajxli opened the pull request:
- Added a few customer API mutations

syfiawoo reviewed (commented):
Though not marked for review, the get_customer_metafields example is wrong for api version 2023-01.
Could you also check the existing examples to make sure they follow the guidelines.
I think we should have examples around customer addresses

Luke-Kav commented on examples/04_local_pickup/01_get_pickup_availability_for_variants/query.graphql:
```diff
@@ -1,25 +1,21 @@
     }
   }
-}
+}
```
Hey @emmajxli - I think the inputs could be dynamic here to keep with the principles of the SFAPI learning kit -  Something like:

```
query GetStoreAvailability($handle: String! $name: String! $value:String!) {
  product(handle: $handle) {
    variantBySelectedOptions(selectedOptions: {name: $name, value: $value}) {
      storeAvailability(first: 1) {
        edges {
          node {
            available
            pickUpTime
            location {
              name
            }
          }
        }
      }
    }
  }
}
```
(with variable file added) 

Also, nitpick, on my PR's, I'd included a link to the docs when there's some requirements they have to fulfill before running the SFAPI query :)

emmajxli replied:
variables.json file added for getStoreAvailability query

Luke-Kav commented on examples/04_local_pickup/03_get_preferred_pickup_location_availability/query.graphql:
```diff
@@ -0,0 +1,20 @@
+      }
+    }
+  }
+}
```
[Similar comment to what I said here](https://github.com/Shopify/storefront-api-learning-kit/pull/26/files#r1086166696)

emmajxli replied:
variables.json file added for getPreferredStoreAvailability query

Luke-Kav commented on examples/07_customers/03_get_customer_metafields/query.graphql:
```diff
@@ -4,19 +4,18 @@
 
 # https://shopify.dev/tutorials/retrieve-metafields-with-storefront-api#expose-metafields-to-the-storefront-api
 
```
Feel nitpicky again here but perhaps could make it a little more explicit that exposting metafields to the SFAPI it's a hard requirement  that you _must_ expose them?  Thoughts?

I only say this as Natalie and I had been doing [this](https://github.com/Shopify/storefront-api-learning-kit/blob/main/examples/05_collections/05_get_all_metafields_for_namespace_in_collection/query.graphql) on our Metafield related PR's.

emmajxli replied:
I modified the explanation of this query and referred "expose_metafield_to_SFAPI" query from metafields folder. Let me know if the new explanation makes sense to you folks :)

Luke-Kav commented on examples/07_customers/07_recover_customer/query.graphql:
```diff
@@ -0,0 +1,14 @@
+# This mutation will send a email to customer to reset password of the customer account.
+# Note that this mutation is throttled by IP.
```
I'd suggest include a point in comments about including [the Shopify-Storefront-Buyer-IP](https://shopify.dev/api/usage/authentication#optional-ip-header) (https://shopify.dev/api/storefront/2023-01/mutations/customerrecover#:~:text=you%20can%20provide%20a%20Shopify%2DStorefront%2DBuyer%2DIP%20instead%20of%20the%20request%20IP.), totally up to you though.

Luke-Kav commented on examples/07_customers/05_create_customer/variables.json:
```diff
@@ -0,0 +1,10 @@
+      "firstName": "John",
+      "lastName": "Smith",
+      "password": "qwerty12345",
+      "phone": "1234567890"
```
This phone number,  I tried it and it would result in `"Enter a valid phone number`.

In the old customer create example Scott used an (🇦🇺  number) which will work but I changed the commit to an 🇳🇿 number which also work 😉

syfiawoo commented on examples/04_local_pickup/03_get_preferred_pickup_location_availability/query.graphql:
```diff
@@ -0,0 +1,20 @@
+# Before sending this request, please make sure your app has unauthenticated_read_product_pickup_locations scope.
+# This query will return the availability of the product of a specific location. You can pass a location ID using preferredLocationId argument to quert the location.
+
+query GetPreferredStoreAvailability @inContext(preferredLocationId: "gid://shopify/Location/65607794710") {
```
```suggestion
query getPreferredStoreAvailability @inContext(preferredLocationId: "gid://shopify/Location/65607794710") {
```

emmajxli commented:
Per discussion above, the following changes have been made:
Local pickup API:
- Changed the query name `GetStoreAvailability` to `getStoreAvailability` and `GetPreferredStoreAvailability` to `getPreferredStoreAvailability` for consistency.
- Changed the folder name `02_get_in-store_pickup_location` to `02_get_nearest_pickup_location` for clarity
- Added variables.json files for `getStoreAvailability` and `getPreferredStoreAvailability` query to make the query more dynamic

Customer API:
- Modified the explanation of get_customer_metafields query to refer "expose_metafield_to_SFAPI" query from metafields folder. Added the link of the query `createMetafieldStorefrontVisibility` for better explanation
- Added the `Shopify-Storefront-Buyer-IP` header explanation & benefit to query `recoverCustomerAccount`

Luke-Kav commented on examples/07_customers/03_get_customer_metafields/query.graphql:
```diff
@@ -1,22 +1,23 @@
+# By default, the Storefront API can't read metafields. To make specific metafields visible in the Storefront API, you need to create a MetafieldStorefrontVisibility record. See metafields/expose_metafield_to_SFAPI query for more details: # https://github.com/Shopify/storefront-api-learning-kit/tree/main/examples/05_collections/05_get_all_metafields_for_namespace_in_collection
 
-# https://shopify.dev/tutorials/retrieve-metafields-with-storefront-api#expose-metafields-to-the-storefront-api
+# If you would like to know more regarding metafields, please check the dev doc #https://shopify.dev/tutorials/retrieve-metafields-with-storefront-api#expose-metafields-to-the-storefront-api
```
[I noticed please was in a few places where we reference the docs](https://github.com/Shopify/storefront-api-learning-kit/search?q=please)
